fix: add more info on case-sensitiveness in search (#1817)
bonitasoft/bonita-doc: modules/api/pages/using-list-and-search-methods.adoc

sob.differentFrom</b>(HumanTaskInstanceSearchDescriptor.ASSIGNEE_ID, myUser.getId());
----

> [#case_sensitivity]

Reviewer: For the record you do not need to explicitly write anchors, they are generated automatically by Antora -> https://docs.asciidoctor.org/asciidoc/latest/macros/xref/#anchors

== Case sensitivity

> WARNING: On Oracle database, search are case-sensitive.

Reviewer 2: ```suggestion
WARNING: On Oracle database, search is case-sensitive.
```

> All Search done using search terms are case-insensitive but not filter.

Reviewer 2: ```suggestion
All Search requests done using **search terms** are case-insensitive but **filtering** IS case-sensitive.
```

> All Search requests done using **search terms** are case-insensitive but **filtering** IS case-sensitive.

Reviewer: It might be because i'm not familiar with the subject, but I'm not sure to understand when I have case-sensitiveness and when I don't: 

- On the warning above, you say that 'On oracle, **search** is case sensitive'
- Then you say that search requests using search terms are case-insensitive but filtering is case sensitive

It's a bit confusing and definitively not straight forward, you start your explanation with an exception and  then you state a rule. 
I suggest you to remove the warning an to simply write two sentences: 
- On most of the database, ...
- On Orcale, ...

Author: I agree. I'll change that!

|===
| Phrase in database | Search string | Matches

| Hello Charles
| `searchBuilder.filter("name", "Hello Charles")`
| yes

| Hello Charles
| `searchBuilder.filter("name", "hello charles")`
| no

| Hello Charles

> | `searchBuilder.searchTerm( "Hello Charles")`

Reviewer 2: ```suggestion
| `searchBuilder.searchTerm("Hello Charles")`
```

| yes

| Hello Charles
| `searchBuilder.searchTerm("hello charles")`
| yes
|===

[#word_based_search]

> 

Reviewer: ```suggestion
```
